src/plugins/tscn/tscnplugin.cpp

```diff
@@ -0,0 +1,735 @@
+/*
+ * Godot 4 Scene Tiled Plugin
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "tscnplugin.h"
+
+#include "grouplayer.h"
+#include "map.h"
+#include "mapobject.h"
+#include "objectgroup.h"
+#include "savefile.h"
+#include "tile.h"
+#include "tilelayer.h"
+
+#include <QCoreApplication>
+#include <QDir>
+#include <QDirIterator>
+#include <QFileInfo>
+#include <QMap>
+#include <QRegularExpression>
+
+#include <iostream>
+#include <stdexcept>
+
+using namespace Tiled;
+using namespace Tscn;
+
+TscnPlugin::TscnPlugin()
+{
+}
+
+// Convenience functions for formatting strings.
+// This is very similar to the vararg version of Qt's QString::arg function,
+// available in Qt 5.14. But as of 2023-01-02, Qt's offline installer is only 5.12.
```

Comment:
It does not look like this vararg overload is ever going to help us here, because it notes "Args can consist of anything that implicitly converts to QString, QStringView or QLatin1StringView." Hence, it will still require numbers to be explicitly converted using QString::number.

I'm personally a bit undecided whether it's really worth adding this overload, rather than just typing QString::number in case of multiple arguments, which I've always done. Especially since this causes additional overhead, due to the `arg` function being called again for each argument.